app/src/main/java/org/sopt/dosopttemplate/AuthViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package org.sopt.dosopttemplate
+
+import android.util.Log
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import org.sopt.dosopttemplate.ServicePool.authService
+import org.sopt.dosopttemplate.request.RequestLoginDto
+import org.sopt.dosopttemplate.request.RequestSignUpDto
+import org.sopt.dosopttemplate.response.ResponseLoginDto
+
+class AuthViewModel : ViewModel() {
+
+    private val _loginSuccess = MutableLiveData<Boolean>()
+    val loginSuccess: MutableLiveData<Boolean>
+        get() = _loginSuccess
+
+    private var _loginResult = MutableLiveData<ResponseLoginDto>()
+    val loginResult: MutableLiveData<ResponseLoginDto>
+        get() = _loginResult
+
+    private val _signUpSuccess = MutableLiveData<Boolean>()
+    val signUpSuccess: MutableLiveData<Boolean>
+        get() = _signUpSuccess
+
+    val username = MutableLiveData("")
+    val password = MutableLiveData("")
+    val nickname = MutableLiveData("")
+    val mbti = MutableLiveData("")
+
+    val isLoginButtonClicked: MutableLiveData<Boolean> = MutableLiveData(false)
+
+    fun onLoginButtonClick() {
+        isLoginButtonClicked.value = true
+    }
+
+    val checkBtnEnabled = MediatorLiveData<Boolean>().apply {
+        addSource(username) { value = isSignUpValid() }
+        addSource(nickname) { value = isSignUpValid() }
+        addSource(password) { value = isSignUpValid() }
+        addSource(mbti) { value = isSignUpValid() }
+        Log.d("signup", "싸인업")
+    }
```

Comment:
```suggestion
    val checkBtnEnabled = MediatorLiveData<Boolean>().apply {
        listOf(username, nickname, password, mbti).forEach { source ->
            addSource(source) { value = isSignUpValid() }
        }
    }
```
이렇게도 작성할 수 있답니다 (취향차이..!!)

Reply:
와웅 코드가 정말 짧아졌네요 ,,! 참고해서 코틀린 문법도 더 공부해볼게요 !

Reply:
엄마 최고 @2zerozu